src/cart/crt_hg_proc.c

```diff
@@ -722,15 +734,19 @@ crt_proc_in_common(crt_proc_t proc, crt_rpc_input_t *data)
 	/* D_DEBUG("in crt_proc_in_common, data: %p\n", *data); */
 
 	if (proc_op != CRT_PROC_FREE) {
-		if (proc_op == CRT_PROC_ENCODE)
+		if (proc_op == CRT_PROC_ENCODE) {
 			rpc_priv->crp_req_hdr.cch_flags = rpc_priv->crp_flags;
+			rc = crt_hlc_send(&rpc_priv->crp_req_hdr.cch_hlc);
+			if (rc) {
+				D_ERROR("crt_hlc_send failed rc: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		}
 		rc = crt_proc_common_hdr(proc, &rpc_priv->crp_req_hdr);
 		if (rc != 0) {
 			D_ERROR("crt_proc_common_hdr failed rc: %d.\n", rc);
 			D_GOTO(out, rc);
 		}
-		if (proc_op == CRT_PROC_DECODE)
-			rpc_priv->crp_flags = rpc_priv->crp_req_hdr.cch_flags;
```

Comment:
hmm, why remove above 2 lines?
it was used when forward corpc from parent to child if I remember correctly.
I'll request a change for this.

Reply:
From my point of view this is dead code path. I add the following in the beginning of this function:
D_ASSERT(proc_op != CRT_PROC_DECODE);
and then run applications. I didn't get this assert. This function is used only for ENCODE.

Could you point me exact case where it's used in DECODE?

Reply:
I see why did not see the DECODE case here.
crt_proc_in_common is the proc func registered to HG, it will be called in two paths:
1) within HG_Forward -> hg_set_input ..., to pack in the request
2) when received the RPC, if user calls HG_Get_input it will call it with DECODE. We used to call HG_Get_input very early phase so that was needed. And I changed that handling to be crt_hg_unpack_header + _unpack_body and removed the direct calling of HG_Get_input. 
So you are right that no DECODE case in our current usage. But for code integrity, I think it is better to keep the above two lines (it will cause bug if we change to directly call HG_Get_input some time later), or comment out it with some additional comments to make it clear.

Reply:
Thanks for explanation. I'd better comment this and put assert in case of logic change.